projects/scripts/adi_board.tcl

```diff
@@ -761,80 +761,47 @@ proc ad_cpu_interrupt {p_ps_index p_mb_index p_name} {
 ###################################################################################################
 ###################################################################################################
 
-# converting a string of ascii characters to a list of corresponding hex values
-proc char2hexlist {str} {
-
-  # string to int list
-  binary scan $str c* int_list
-
-  # declare empty list
-  set hex_list {}
-
-  # convert int to hex and append to list
-  foreach i $int_list {
-    lappend hex_list [format %0.2X [expr {$i & 0xFF}]]
+proc stringtohex {str blocksize} {
+  binary scan $str H* hex
+  while {[string length $hex] < $blocksize * 2} {
```

Comment:
A loop-less solution would be "set hex [format %-${blocksize}s $hex". This will pad the string, to have a length of $blocksize, with spaces.

Reply:
done